### ISSUE TITLE
Kucoin fetch OHLCV

### ISSUE THREAD

**Issue by kz0** [ccxt/ccxt#2677]
- OS: Ubuntu LTS 16.04
- Programming Language version: Python 3.5.2
- CCXT version: ccxt-1.13.48
- Exchange: Kucoin
- Method: fetch_ohlcv

When trying to fetch candles i receive the following error:

```
---------------------------------------------------------------------------
ExchangeError                             Traceback (most recent call last)
<ipython-input-17-1854c4f6047e> in <module>()
----> 1 candles = exchange.fetch_ohlcv('ETH/BTC', timeframe='1m')
      2 
      3 print(candles)

~/kucoin-bot/lib/python3.5/site-packages/ccxt/kucoin.py in fetch_ohlcv(self, symbol, timeframe, since, limit, params)
    569             iso8601 = self.iso8601(timestamp)
    570         order = {
--> 571             'info': response,
    572             'id': orderId,
    573             'timestamp': timestamp,

~/kucoin-bot/lib/python3.5/site-packages/ccxt/base/exchange.py in request(self, path, api, method, params, headers, body)
    295             time.sleep(delay / 1000.0)
    296 
--> 297     def fetch2(self, path, api='public', method='GET', params={}, headers=None, body=None):
    298         """A better wrapper over request for deferred signing"""
    299         if self.enableRateLimit:

~/kucoin-bot/lib/python3.5/site-packages/ccxt/base/exchange.py in fetch2(self, path, api, method, params, headers, body)
    292         elapsed = now - self.lastRestRequestTimestamp
    293         if elapsed < self.rateLimit:
--> 294             delay = self.rateLimit - elapsed
    295             time.sleep(delay / 1000.0)
    296 

~/kucoin-bot/lib/python3.5/site-packages/ccxt/base/exchange.py in fetch(self, url, method, headers, body)
    367             self.raise_error(RequestTimeout, method, url, e)
    368 
--> 369         except TooManyRedirects as e:
    370             self.raise_error(ExchangeError, url, method, e)
    371 

~/kucoin-bot/lib/python3.5/site-packages/ccxt/kucoin.py in handle_errors(self, code, reason, url, method, headers, body, response)
    650         #     }
    651         #     return response
--> 652         #
    653         return self.privatePostCancelOrder(self.extend(request, params))
    654 

~/kucoin-bot/lib/python3.5/site-packages/ccxt/kucoin.py in throw_exception_on_error(self, response)
    618         if symbol is None:
    619             raise ExchangeError(self.id + ' cancelOrder requires a symbol')
--> 620         self.load_markets()
    621         market = self.market(symbol)
    622         request = {

ExchangeError: kucoin: malformed response: {"v":[0.01,0.0,0.06189139,2.2941378,0.0,0.287,0.02,0.0,0.0,0.305582,11.78552736,15.24708459,48.85852063,179.38577081,137.06173522,228.09247984,65.4742495,128.12426182,119.73331385,170.13524704,252.48273233,163.20710658,145.00064661,190.75155323,205.9017018,168.0392611,179.63432629,191.91203599,148.18418477,176.55350444,182.30186787,147.10771694,379.70376737,399.37174786,236.63000946,77.16153932,160.41625359,317.00296832,539.50333716,423.29371774,505.49392942,570.34425752,1291.95173132,902.35187999,761.00337721,1091.20621214,509.82252659,810.78101354,614.85552915,523.16712821,545.45138822,420.0733373,503.89284083,1409.32877175,681.8486349,852.2847961,1202.85549869,1056.77550522,1038.8090564,900.18251727,1247.28636632,1250.463834,1089.832576,580.989675,801.084697,642.110104,649.165132,774.816855,773.405285,608.382187,679.843069,632.609185,556.058927,421.650668,466.700553,317.005713,431.024491,285.19121,372.143092,408.490859,490.78515028,583.325076,502.372803,422.430824,261.46257,314.474531,630.517683,409.994052,408.790391,340.640618,330.526226,291.629366,358.245738,729.832622,908.872659,1002.383199,1260.618566,1217.783748,1131.245991,1250.23834,1544.510377,2062.231804,2946.34813,5836.378499,6010.945455,4906.840673,5733.434794,8297.684722,7396.716829,9297.550935,9753.946717,8618.865748,11698.202674,11617.323103,16611.903405,18305.172843,15943.160983,12809.146995,11275.591527,8284.510503,8096.328841,18312.890717,12192.657311,11227.000662,9310.845637,10453.215711,8835.793525,6733.077476,7465.88247,5542.735967,8857.506119,9243.52841,8004.659945,11229.72908,13104.637814,11209.352546,10986.355003,11054.458904,9797.584176,8793.544674,4614.846085,4388.805038,9397.171767,7785.549428,5876.616026,5860.014093,6290.580175,4677.445216,4725.975948,4247.020825,4217.920564,5216.24249,5025.016965,3965.003987,5576.132955,4237.231139,5316.973236,6383.767602,4457.025378,3530.403293,3242.968576,2408.237362,2920.651661,4375.661527,3315.385096,3645.268695,4767.410673,3941.774659,3527.983002,3296.46619,3722.328882,5332.04766,5164.708655,4855.429975,3214.708719,2026.576488,2767.9174678,2557.7930761,3211.6078812,5029.8970801,3107.5074671,2440.2505996,2864.2769704,5249.5345557,3832.8632323,3483.8512763,4709.9380733,3748.700622,2916.0364586,2684.1630763,4575.6303908,4661.6043424,3670.3449355,5605.61117,4472.654222,3897.9317627,2105.2465661,3345.1688653,4430.7172957,4665.2406997,4288.4435687,2666.7012573,2727.0710269,2693.2090322,3096.7232867,5435.1104951,5940.9968212,3992.3913609,4574.9885788,4377.8964241,2617.527858,4205.8879087,3344.9666756,3441.6988189,4523.812995,4710.9316327,4906.5535683,6084.8872731,6052.9117109,6752.5034605,8258.591822,7608.994097,1535.9552814],"h":[0.0686,null,0.0686,0.0686,null,0.00968,0.07849,null,null,0.07846,0.074481,0.0776,0.0739404,0.07267154,0.07196002,0.071104,0.0705913,0.06993194,0.07051111,0.0699099,0.072722,0.0729,0.071,0.0674711,0.06607,0.06507304,0.06199,0.06423373,0.062,0.0649,0.0622222,0.06096,0.06229,0.05863364,0.05459999,0.052,0.0515,0.05519798,0.05519,0.054967,0.05298937,0.0524895,0.055,0.05518694,0.05149877,0.05018996,0.04646275,0.04455092,0.04396278,0.043,0.04109178,0.04350831,0.045,0.04768091,0.04676908,0.04644213,0.054,0.05497799,0.0528,0.05200001,0.05,0.045,0.04468,0.052,0.0488,0.04999,0.0455,0.05,0.053,0.06,0.05694694,0.05292638,0.05094899,0.04979894,0.04723998,0.04699999,0.045979,0.04435,0.04259759,0.04257898,0.04190001,0.0368,0.02999999,0.03296074,0.0368,0.03229999,0.03269999,0.04099999,0.04649699,0.04489096,0.04445555,0.04729999,0.03923783,0.04729982,0.04999999,0.0535,0.05485278,0.05025,0.05069999,0.05321303,0.0556,0.0512,0.052,0.0572135,0.06898162,0.05684014,0.06149,0.06796,0.0658,0.06597,0.073,0.06393001,0.06965,0.07999998,0.08771829,0.10298899,0.0975,0.092957,0.09499999,0.101,0.10291902,0.10271,0.09718255,0.0932,0.0927201,0.101,0.0929,0.092974,0.09393,0.09238,0.0955,0.09625028,0.09723562,0.10479788,0.10839974,0.12,0.11250364,0.116,0.12420999,0.10980699,0.10887999,0.10369705,0.10225774,0.10580747,0.1026,0.102,0.102,0.1020199,0.1025,0.09982785,0.0999999,0.098876,0.0967,0.09499992,0.09168197,0.08952182,0.08741696,0.082678,0.081078,0.08579981,0.08669,0.087971,0.089485,0.0889,0.08332097,0.08347,0.08181947,0.0795,0.07649162,0.07762603,0.07619993,0.0767899,0.07719146,0.07782333,0.07990996,0.0795,0.07929998,0.07751989,0.0772,0.0761,0.0762,0.0739679,0.07347719,0.067844,0.06548877,0.06497995,0.06424712,0.06266799,0.06218896,0.061412,0.06242998,0.06110584,0.05856637,0.05778979,0.05684698,0.0589322,0.0573574,0.0567,0.0559838,0.056408,0.05635621,0.0567,0.05628099,0.05656341,0.059941,0.05975891,0.0615,0.0633,0.06449999,0.064906,0.06393439,0.064165,0.066654,0.06443407,0.06568686,0.07,0.07110297,0.06982086,0.07245001,0.07425057,0.07587,0.07178671,0.07190952],"s":"ok","l":[0.0686,null,0.0686,0.000685,null,0.00968,0.07849,null,null,0.07846,0.07085263,0.07088,0.0676642,0.06666,0.066452,0.06428,0.000685,0.0633,0.0673892,0.06666,0.063,0.068011,0.06535221,0.05957269,0.0604,0.056,0.05069526,0.05391732,0.05527717,0.05842801,0.05385001,0.05391432,0.048,0.05000101,0.04718516,0.046,0.0465,0.04600002,0.05028808,0.05004004,0.0481001,0.0451,0.0475,0.0485097,0.04933001,0.04320112,0.0380025,0.03700001,0.0387118,0.03900099,0.03892302,0.04020869,0.04000801,0.0380076,0.0430121,0.04390878,0.04480001,0.04660939,0.04597001,0.0475,0.043,0.04030397,0.04060002,0.04050301,0.04320223,0.04350001,0.04420001,0.044,0.0480009,0.050297,0.04951,0.046789,0.0475,0.043509,0.04260001,0.04289517,0.0417,0.0412,0.04020302,0.039116,0.034813,0.02692151,0.00493776,0.02610302,0.03001002,0.02711191,0.02630001,0.03061222,0.03710026,0.03450001,0.03511808,0.035105,0.03513218,0.03763003,0.04349201,0.046323,0.04686133,0.046,0.0464,0.04746101,0.04785094,0.04559202,0.04641801,0.04920004,0.05009702,0.05210004,0.05290002,0.05540301,0.05550001,0.05758001,0.0602,0.0542105,0.0575905,0.0641,0.07,0.08001454,0.08300101,0.08560002,0.086085,0.092864,0.095278,0.08603153,0.085,0.07777,0.088222,0.08775114,0.08870001,0.09000508,0.09119005,0.089,0.09040002,0.09225981,0.09400011,0.09532502,0.10292599,0.10309,0.10490478,0.107801,0.104,0.0966,0.102,0.1,0.094,0.09526003,0.09828651,0.09625939,0.0995001,0.09911117,0.0985941,0.0965125,0.09731002,0.09450001,0.091,0.08861182,0.08657998,0.086022,0.08153901,0.076936,0.07785701,0.080207,0.082825,0.08410021,0.08632495,0.08239903,0.08034008,0.08097,0.07811111,0.075637,0.07428633,0.07457213,0.073,0.074,0.07323,0.074166,0.07602613,0.07708524,0.07461398,0.07430368,0.07483802,0.07237991,0.0725114,0.07151,0.065355,0.06191086,0.06184,0.05999998,0.0614,0.06082985,0.0603008,0.0600141,0.06013401,0.0573,0.056593,0.054427,0.05352851,0.0551166,0.0559,0.0545609,0.05368385,0.053696,0.053702,0.05559,0.05514,0.05514,0.05601839,0.05785108,0.05847,0.0600295,0.0594301,0.06199879,0.06170101,0.0625022,0.06290013,0.06323147,0.06341289,0.06500007,0.06732682,0.06648,0.06944,0.07129724,0.06688515,0.06765412,0.07089142],"o":[0.0686,null,0.0686,0.0686,null,0.00968,0.07849,null,null,0.07846,0.0732,0.074502,0.07261665,0.07145844,0.0699,0.06875128,0.069162,0.0633,0.06877341,0.0699099,0.06833492,0.069689,0.06817879,0.0656,0.06149485,0.06173419,0.0581,0.05637191,0.06166204,0.06082745,0.06096437,0.05762373,0.06051302,0.05436103,0.05380021,0.0498,0.04790017,0.05059897,0.055,0.052705,0.05054053,0.05001999,0.05027037,0.05149996,0.04999989,0.05001989,0.043211,0.0380025,0.0394,0.0428,0.03980864,0.04089181,0.04167,0.04035314,0.0437,0.04512329,0.04565001,0.048,0.04925,0.0509,0.0487,0.04320999,0.04201649,0.04261272,0.04679995,0.04440001,0.04485452,0.044425,0.04959698,0.05140001,0.05545,0.05141973,0.04927861,0.04765289,0.0437905,0.04514501,0.04408954,0.04221,0.04197656,0.0414,0.0403495,0.03489264,0.02829999,0.02806601,0.03183626,0.03229999,0.02799993,0.03196791,0.03902017,0.04406663,0.03627254,0.03875,0.03630001,0.0382425,0.044288,0.0469,0.0504205,0.04899,0.04938,0.04833002,0.0513,0.04984997,0.04732775,0.0495301,0.05110064,0.05359999,0.05467,0.0566,0.0628298,0.0621997,0.064112,0.06080772,0.061,0.065,0.07529893,0.0827,0.09236101,0.0895995,0.086242,0.0938002,0.099398,0.09617392,0.08819821,0.087825,0.08881998,0.090962,0.089014,0.090704,0.092554,0.092,0.09149999,0.09459002,0.09579471,0.09532502,0.10303929,0.10618744,0.1067475,0.10857976,0.11550201,0.10530697,0.10727989,0.102477,0.10167075,0.0964081,0.10151205,0.09914992,0.100406,0.10109789,0.10044,0.099,0.09807523,0.09821998,0.09516777,0.09400002,0.08939405,0.087847,0.087268,0.082362,0.08052503,0.0805,0.0855,0.0845255,0.087568,0.088819,0.082557,0.08158029,0.08163508,0.07901211,0.076018,0.075712,0.07538785,0.07584506,0.074681,0.07646299,0.0767587,0.077807,0.07890101,0.07499598,0.076684,0.07589301,0.074648,0.073702,0.07289198,0.06608716,0.064,0.062524,0.0638701,0.061825,0.06150945,0.060535,0.06118613,0.061053,0.05797999,0.057428,0.054632,0.056015,0.05710001,0.05638604,0.05558069,0.054366,0.053702,0.05605887,0.055885,0.055472,0.05616077,0.059425,0.05867372,0.0604358,0.06250985,0.063852,0.06231889,0.06322405,0.06349812,0.06365632,0.063715,0.06516199,0.0689188,0.06735,0.06978168,0.071782,0.07392279,0.06798147,0.07144801],"c":[0.0686,null,0.0686,0.000685,null,0.00968,0.07849,null,null,0.07846,0.074481,0.07263931,0.06911555,0.06900513,0.06888888,0.06855369,0.06315405,0.0680394,0.0684946,0.06810001,0.07054397,0.0697518,0.068,0.06151319,0.06288247,0.0591156,0.05441386,0.06183128,0.06199508,0.06206485,0.05776834,0.05782218,0.05629464,0.0546,0.04800021,0.04998711,0.05059896,0.05400001,0.05270501,0.05199998,0.05001999,0.05027037,0.05149999,0.04943956,0.04990001,0.04320999,0.0380025,0.03944999,0.0415916,0.0399,0.04089181,0.0421,0.041,0.0437,0.04511421,0.04563334,0.048,0.04934013,0.0509,0.0487,0.04312006,0.04239999,0.04261265,0.04679995,0.0445,0.04485451,0.044425,0.04941225,0.05140001,0.05545,0.05121236,0.04886036,0.04779976,0.04399999,0.04514501,0.0445,0.042015,0.04219999,0.04139998,0.0403495,0.035,0.029,0.02853301,0.03150001,0.03239999,0.02767959,0.03090607,0.03902017,0.04489699,0.0369,0.03831205,0.03664995,0.0385,0.045,0.04748387,0.05000001,0.04888349,0.04938,0.04833001,0.0513,0.04927001,0.047466,0.0495301,0.0511006,0.05359999,0.05467,0.0566,0.0628298,0.06219971,0.064111,0.06050752,0.061,0.06501,0.07529893,0.0827,0.09236101,0.0895995,0.086242,0.09428393,0.0994,0.09617392,0.08819821,0.08782999,0.08881998,0.09096006,0.089006,0.090704,0.0929605,0.092,0.0913086,0.09458,0.09579471,0.09575185,0.102927,0.10618745,0.10692798,0.10857976,0.115506,0.10540693,0.10727983,0.10247702,0.101748,0.0964081,0.10151205,0.09914993,0.10040789,0.1014,0.10026718,0.09919299,0.09869,0.098,0.09543093,0.09399798,0.08939408,0.08819398,0.086889,0.082362,0.08068426,0.08070294,0.0855,0.0845255,0.087299,0.08865,0.08283289,0.08185,0.08181742,0.07909096,0.076018,0.07571197,0.07538898,0.07573208,0.074681,0.07646299,0.0767587,0.07815202,0.07890101,0.07499598,0.076684,0.07589301,0.0746,0.07356578,0.073045,0.06608716,0.064,0.062522,0.0638701,0.061963,0.06150945,0.060535,0.06118613,0.061055,0.0580353,0.05724701,0.05472001,0.056015,0.05710001,0.05638606,0.055517,0.05439799,0.053702,0.05605689,0.05590301,0.05554975,0.05616077,0.05945442,0.05867372,0.0604358,0.06248749,0.063887,0.062358,0.06322405,0.06349812,0.0637124,0.06371525,0.065092,0.06911999,0.06735,0.06978168,0.07167096,0.074108,0.06798147,0.07144801,0.071414],"t":[1505559600,1505646000,1505732400,1505818800,1505905200,1505991600,1506078000,1506164400,1506250800,1506337200,1506423600,1506510000,1506596400,1506682800,1506769200,1506855600,1506942000,1507028400,1507114800,1507201200,1507287600,1507374000,1507460400,1507546800,1507633200,1507719600,1507806000,1507892400,1507978800,1508065200,1508151600,1508238000,1508324400,1508410800,1508497200,1508583600,1508670000,1508756400,1508842800,1508929200,1509015600,1509102000,1509188400,1509274800,1509361200,1509447600,1509534000,1509620400,1509706800,1509793200,1509879600,1509966000,1510052400,1510138800,1510225200,1510311600,1510398000,1510484400,1510570800,1510657200,1510743600,1510830000,1510916400,1511002800,1511089200,1511175600,1511262000,1511348400,1511434800,1511521200,1511607600,1511694000,1511780400,1511866800,1511953200,1512039600,1512126000,1512212400,1512298800,1512385200,1512471600,1512558000,1512644400,1512730800,1512817200,1512903600,1512990000,1513076400,1513162800,1513249200,1513335600,1513422000,1513508400,1513594800,1513681200,1513767600,1513854000,1513940400,1514026800,1514113200,1514199600,1514286000,1514372400,1514458800,1514545200,1514631600,1514718000,1514804400,1514890800,1514977200,1515063600,1515150000,1515236400,1515322800,1515409200,1515495600,1515582000,1515668400,1515754800,1515841200,1515927600,1516014000,1516100400,1516186800,1516273200,1516359600,1516446000,1516532400,1516618800,1516705200,1516791600,1516878000,1516964400,1517050800,1517137200,1517223600,1517310000,1517396400,1517482800,1517569200,1517655600,1517742000,1517828400,1517914800,1518001200,1518087600,1518174000,1518260400,1518346800,1518433200,1518519600,1518606000,1518692400,1518778800,1518865200,1518951600,1519038000,1519124400,1519210800,1519297200,1519383600,1519470000,1519556400,1519642800,1519729200,1519815600,1519902000,1519988400,1520074800,1520161200,1520247600,1520334000,1520420400,1520506800,1520593200,1520679600,1520766000,1520852400,1520938800,1521025200,1521111600,1521198000,1521284400,1521370800,1521457200,1521543600,1521630000,1521716400,1521802800,1521889200,1521975600,1522062000,1522148400,1522234800,1522321200,1522407600,1522494000,1522580400,1522666800,1522753200,1522839600,1522926000,1523012400,1523098800,1523185200,1523271600,1523358000,1523444400,1523530800,1523617200,1523703600,1523790000,1523876400,1523962800,1524049200,1524135600,1524222000,1524308400,1524394800,1524481200,1524567600,1524654000,1524740400]}
```

**Comment by kroitor** [comment]
@kz0 hi! Sorry for the long wait... Can you plz show a minimal snippet of your code to reproduce the issue? (including the exchange class instantiation)

**Comment by kz0** [comment]
No problem. I was using the jupyter notebook with this simple code:

```
import ccxt
from secrets import api_key, api_secret

exchange = ccxt.kucoin({
        'apiKey': api_key,
        'secret': api_secret, 
        'nonce': ccxt.Exchange.milliseconds,
    })


exchange.load_markets()

candles = exchange.fetch_ohlcv('ETH/BTC', timeframe='1m')

print(candles)
```

**Comment by kz0** [comment]
Also i just noticed fetching the orderbook using the code:

`orderbook = exchange.fetch_order_book(symbol, {'limit': 100})`

only returns the top 6 bids.

**Comment by kroitor** [comment]
> Nevermind, i wasn't properly sending the params.

The signature of fetch_order_book is:

```Python
fetch_order_book(symbol, limit, params)
```

The second `limit` argument is the unified way for limiting the orderbook with that fetch call.

So, these examples are equivalent for Kucoin:

```Python
# unified, will work for other exchanges
orderbook = exchange.fetch_order_book('ETH/BTC', 100)
```

```Python
# non-unified, Kucoin-only (exchange-specific)
orderbook = exchange.fetch_order_book('ETH/BTC', params={'limit': 100}) 
```

```Python
# params from 3rd argument override the limit from the second argument in exchange-specific way
orderbook = exchange.fetch_order_book('ETH/BTC', 100, {'limit': 50}) 
```

**Comment by kroitor** [comment]
@kz0 the above symptoms can happen if you don't use proper rate limiting, this issue with last_json_response has been addressed already and should be resolved by updating to version 1.16.41. Let us know if updating + adding `{'enableRateLimit': True}` to constructor params does not help.
